### PR TITLE
Support Sigmoid for IE/DML

### PR DESCRIPTION
**Pull request by miaobin**
I have tested IE on Linux and DML on Windows.
@fujunwei @mingmingtasd PTAL, thanks.

### CODE REVIEW

**Review by huningxin — APPROVED** [review]
Thanks @miaobin , could you please the test results?

**Comment by huningxin** [comment]
And please also help solve the conflicting files. Thanks.

File: src/tests/end2end/SigmoidTests.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/tests/WebnnTest.h"
+
+class SigmoidTests : public WebnnTest {};
+
+TEST_F(SigmoidTests, Sigmoid_1D) {
```
Comment (mingmingtasd) [inline src/tests/end2end/SigmoidTests.cpp:19]: Please name as Sigmoid1D to keep the format consistent.

File: src/tests/end2end/SigmoidTests.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    ASSERT_TRUE(graph);
+    const std::vector<float> inputData = {-1, 0, 1};
+    const ml::Input input = {inputData.data(), inputData.size() * sizeof(float)};
+    const ml::Result result = utils::AwaitCompute(graph, {{"a", input}}).Get("b");
+    EXPECT_TRUE(utils::CheckShape(result, {3}));
+    const std::vector<float> expectedData = {0.26894143, 0.5, 0.7310586};
+    EXPECT_TRUE(utils::CheckValue(result, expectedData));
+}
+
+TEST_F(SigmoidTests, Sigmoid_3D) {
```
Comment (mingmingtasd) [inline src/tests/end2end/SigmoidTests.cpp:33]: Sigmoid3D

**Review by mingmingtasd — APPROVED** [review]
LGTM, thanks! @miaobin

**Comment by miaobin** [comment]
Add the test results and olve the conflicting files. @huningxin PTLA, thanks!

IE: webnn_end2end_tests --gtest_filter=Sigmoid*
Note: Google Test filter = Sigmoid*
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from SigmoidTests
[ RUN      ] SigmoidTests.Sigmoid_1D
[       OK ] SigmoidTests.Sigmoid_1D (11 ms)
[ RUN      ] SigmoidTests.Sigmoid_3D
[       OK ] SigmoidTests.Sigmoid_3D (5 ms)
[----------] 2 tests from SigmoidTests (16 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test suite ran. (16 ms total)
[  PASSED  ] 2 tests.

DML: webnn_end2end_tests.exe --gtest_filter=Sigmoid*
Note: Google Test filter = Sigmoid*
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from SigmoidTests
[ RUN      ] SigmoidTests.Sigmoid_1D
[       OK ] SigmoidTests.Sigmoid_1D (419 ms)
[ RUN      ] SigmoidTests.Sigmoid_3D
[       OK ] SigmoidTests.Sigmoid_3D (34 ms)
[----------] 2 tests from SigmoidTests (453 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test suite ran. (468 ms total)
[  PASSED  ] 2 tests.

**Pull request merged**